documentation/docs/concepts/decentralized_identifiers/update.mdx

```diff
@@ -46,363 +50,244 @@ methods allow them to be used by more than one verification relationship.
-         }
-      ],
-      "capabilityDelegation":[
-         "did:iota:6T4PRHWp7bNsKaBWr1gVtUBQfLaxAKqKGAeJWFBZkMyf#my-next-key"
-      ],
-      "capabilityInvocation":[
-         {
-            "id":"did:iota:6T4PRHWp7bNsKaBWr1gVtUBQfLaxAKqKGAeJWFBZkMyf#sign-0",
-            "controller":"did:iota:6T4PRHWp7bNsKaBWr1gVtUBQfLaxAKqKGAeJWFBZkMyf",
-            "type":"Ed25519VerificationKey2018",
-            "publicKeyMultibase":"z5k7vzMVuXXj8MJDcfzP2owvc8xKBA6BBsAkFf1GSNu2X"
-         },
-         "did:iota:6T4PRHWp7bNsKaBWr1gVtUBQfLaxAKqKGAeJWFBZkMyf#my-next-key"
-      ]
-   },
-   "meta":{
-      "created":"2022-04-13T09:27:48Z",
-      "updated":"2022-04-13T09:28:23Z"
-   }
+  "doc": {
+    "id": "did:iota:rms:0x7408d40ceaf44b83ad03b0e05f5a5fe21adce0b47a133f91657efe4e41619a06",
+    "verificationMethod": [
+      {
+        "id": "did:iota:rms:0x7408d40ceaf44b83ad03b0e05f5a5fe21adce0b47a133f91657efe4e41619a06#key-1",
+        "controller": "did:iota:rms:0x7408d40ceaf44b83ad03b0e05f5a5fe21adce0b47a133f91657efe4e41619a06",
+        "type": "Ed25519VerificationKey2018",
+        "publicKeyMultibase": "z8ogKEagUH8swGf9vJNwsLQk79HSm2rkbQjfsMuGQqBjG"
+      }
+    ],
+    "authentication": ["did:iota:rms:0x7408d40ceaf44b83ad03b0e05f5a5fe21adce0b47a133f91657efe4e41619a06#key-1"]
+  },
+  "meta": {
+    "created": "2022-09-11T19:21:47Z",
+    "updated": "2022-09-11T19:21:47Z"
+  }
 }
 ```
 
+Notice that these changes to the document are not published yet. This will be done after adding the service.
+
 ### Adding a Service
-Similar to verification methods, services can be added to a DID Document.
+
 <Tabs groupId="code-examples">
 <TabItem value="Rust" label="Rust">
 
 ```rust
-account
-    .update_identity()
-    .create_service()
-    .fragment("my-service-1")
-    .type_("MyCustomService")
-    .endpoint(Url::parse("https://example.com")?)
-    .apply()
-    .await?;
+// Add a new Service.
+let service: StardustService = Service::from_json_value(json!({
+    "id": document.id().to_url().join("#linked-domain")?,
+    "type": "LinkedDomains",
+    "serviceEndpoint": "https://iota.org/"
+}))?;
+assert!(document.insert_service(service));
+document.metadata.updated = Some(Timestamp::now_utc());
 ```
 
 </TabItem>
 <TabItem value="node" label="Node.js">
 
 ```js
-await account.createService({
-   fragment: "my-service-1",
-   type: "MyCustomService",
-   endpoint: "https://example.com"
-})
+// Add a new Service.
+const service: StardustService = new StardustService({
+  id: did.join("#linked-domain"),
+  type: "LinkedDomains",
+  serviceEndpoint: "https://iota.org/"
+});
+document.insertService(service);
+document.setMetadataUpdated(Timestamp.nowUTC());
 ```
+
 </TabItem>
 </Tabs>
 
-In JavaScript, the endpoint property type is a string, this must be a valid URL, otherwise an error will be thrown.
+In JavaScript, the `serviceEndpoint` property type is a string, this must be a valid URL, otherwise an error will be thrown.
```

Comment:
Both Rust and Node.js use the JSON constructor, so this doesn't apply. Also we support sets and maps of endpoints now, so it's not entirely accurate, maybe linking out would be better.

```suggestion
```